src/gui/widgets/Graph.cpp

```diff
@@ -200,48 +200,29 @@ void Graph::drawLineAt( int _x, int _y, int _lastx )
 
 	float range = minVal - maxVal;
 	float val = ( _y*range/( height()-5 ) ) + maxVal;
-	
-	int sample_begin, sample_end;
-	float lastval;
-	float val_begin, val_end;
-	
-	if (_lastx > _x)
-	{
-		sample_begin = (int)((_x) * xscale);
-		sample_end = (int)ceil((_lastx+1) * xscale);
-		lastval = model() -> m_samples[ (int)( sample_end - 1 ) ];
-		val_begin = val;
-		val_end = lastval;
 
-	}
-	else
-	{
-		sample_begin = (int)(_lastx * xscale);
-		sample_end =  (int)ceil((_x+1) * xscale);
-		lastval = model() -> m_samples[ (int)( sample_begin ) ];
-		val_begin = lastval;
-		val_end = val;
-		
-	}
-	
-	// calculate line drawing variables
-	int linelen = sample_end - sample_begin;
-	if (linelen == 1)
-	{
-		val_begin = val;
-	}
+	const auto pastX = _lastx > _x;
```

Comment:
I don't know if this is an accurate name, but I couldn't think of anything else at the time 😂

Reply:
It works well enough, lol. My only hope is that whatever I did here doesn't cause a bug :rofl: (also just saw the parentheses issue you caught for this, good eye)

Reply:
I don't really like how there's a ternary operator for every statement now - seems less efficient than just a single if-else like before.

Reply:
> I don't really like how there's a ternary operator for every statement now - seems less efficient than just a single if-else like before.

Yeah, I can see that now. I reverted it back to the original.